### ISSUE TITLE
-Wasm-operand-widths in arch/arm64

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#349]
```
In file included from arch/arm64/kernel/asm-offsets.c:22:
In file included from ./include/linux/sched.h:14:
In file included from ./include/linux/pid.h:5:
In file included from ./include/linux/rculist.h:11:
In file included from ./include/linux/rcupdate.h:39:
In file included from ./include/linux/irqflags.h:16:
./arch/arm64/include/asm/irqflags.h:50:10: warning: value size does not match register size specified by the constraint and modifier [-Wasm-operand-widths]
                : "r" (GIC_PRIO_IRQON)
                       ^
./arch/arm64/include/asm/ptrace.h:39:25: note: expanded from macro 'GIC_PRIO_IRQON'
#define GIC_PRIO_IRQON          0xf0
                                ^
./arch/arm64/include/asm/irqflags.h:46:44: note: use constraint modifier "w"
                "msr_s  " __stringify(SYS_ICC_PMR_EL1) ",%0\n"
                                                         ^~
                                                         %w0
./arch/arm64/include/asm/alternative.h:286:29: note: expanded from macro 'ALTERNATIVE'
        _ALTERNATIVE_CFG(oldinstr, newinstr, __VA_ARGS__, 1)
                                   ^
./arch/arm64/include/asm/alternative.h:88:30: note: expanded from macro '_ALTERNATIVE_CFG'
        __ALTERNATIVE_CFG(oldinstr, newinstr, feature, IS_ENABLED(cfg), 0)
                                    ^
./arch/arm64/include/asm/alternative.h:76:2: note: expanded from macro '__ALTERNATIVE_CFG'
        newinstr "\n"                                                   \

```

From CI (this spam caused the build to fail): https://travis-ci.com/ClangBuiltLinux/continuous-integration/jobs/175961058

Introduced by this commit: https://git.kernel.org/pub/scm/linux/kernel/git/arm64/linux.git/commit/?id=4a503217ce37e1f4f3d9b681bbcbbac103776bf1

I assume they'll want something similar to https://git.kernel.org/pub/scm/linux/kernel/git/arm64/linux.git/commit/?id=1b57ec8c75279b873639eb44a215479236f93481

I'll report it after work unless someone beats me to it.

**Comment by nickdesaulniers** [comment]
yep, we've seen these kind of issues before; Clang is strict on the `%w0` vs `%x0` convention.

**Comment by nathanchance** [comment]
Reported on LKML: https://lore.kernel.org/lkml/20190208043543.GA5040@archlinux-ryzen/

**Comment by nathanchance** [comment]
Fix accepted, will close this when it hits next: https://git.kernel.org/pub/scm/linux/kernel/git/arm64/linux.git/commit/?id=81dc19231d101c027aec410416574c946ed2f2e7

**Comment by nathanchance** [comment]
This was fixed in `next-20190211`.
